command/push.py

```diff
@@ -1,8 +1,35 @@
+import base64
+import os
+import requests
+import tarfile
+
 class PushCmd():
     name = 'push'
 
     def __init__(self):
         pass
         
-    def push(self):
-        print("Hello, world. The push feature is currently under construction.")
+    def push(self, bundle_dir, namespace, repository, release, auth_token):
+        base64_bundle = self.__create_base64_bundle(bundle_dir, repository)
+        self.__push_to_registry(namespace, repository, release, base64_bundle, auth_token)
+
+    def __create_base64_bundle(self, bundle_dir, repository):
+        tarfile_name = "%s.tar.gz" % repository
+        with tarfile.open(tarfile_name, "w:gz") as tar:
```

Comment:
Yeah, this seems very useful. I'll take a look at it.

Reply:
I took a look at this, and it seems like this will be useful, but it should be called ahead of this method to create the temporary directory that this method just accesses. (I.e. read the temporary bundle created by the build method, write it to a temporary directory. Then write the tarball to the same temp directory.) It will be addressed in a separate commit.